File: TrinitySQL.Blueprint/Server/TestInfrastructure/TestDatabaseBuilder.cs
using Microsoft.EntityFrameworkCore;
using TrinitySQL.Server.Data;
using TrinitySQL.Server.Infrastructure;

namespace TrinitySQL.Blueprint.Server.TestInfrastructure;

public class TestDatabaseBuilder
{
    public TestDatabase Build()
    {
        var dbPath = Path.Combine(Path.GetTempPath(), $"theater_sales_test_{Guid.NewGuid()}.db");
        var connectionString = $"Data Source={dbPath}";

        var initializer = new DatabaseInitializer(connectionString);
        initializer.Initialize();
        initializer.SeedData();

        var optionsBuilder = new DbContextOptionsBuilder<TheaterSalesContext>();
        optionsBuilder.UseSqlite(connectionString);
        
        var context = new TheaterSalesContext(optionsBuilder.Options);
        var serviceConfig = new ServiceConfiguration(context);
        
        return new TestDatabase(context, serviceConfig.QueryDispatcher, dbPath);
    }
}

File: TrinitySQL.Blueprint/Device/TestInfrastructure/TestDatabaseBuilder.ts
import {DatabaseInitializer, TheaterSalesContext} from '@device/Data'
import {TestDatabase} from './TestDatabase'
import * as path from 'path'
import {randomUUID} from 'crypto'

export class TestDatabaseBuilder
{
  build(): TestDatabase
  {
    const dbPath = this.generateDatabasePath()
    const connectionString = dbPath
    
    this.initializeDatabase(connectionString)
    
    const context = new TheaterSalesContext(connectionString)
    
    return new TestDatabase(context, dbPath)
  }
  
  private generateDatabasePath(): string
  {
    const tempDir = require('os').tmpdir()
    const uniqueId = randomUUID()
    return path.join(tempDir, `theater_sales_test_${uniqueId}.db`)
  }
  
  private initializeDatabase(connectionString: string): void
  {
    const initializer = new DatabaseInitializer(connectionString)
    initializer.Initialize()
    initializer.SeedData()
  }
}